business/checkers/destinationrules/no_dest_checker.go

```diff
@@ -21,33 +21,67 @@ func (n NoDestinationChecker) Check() ([]*models.IstioCheck, bool) {
 	if host, ok := n.DestinationRule.GetSpec()["host"]; ok {
 		if dHost, ok := host.(string); ok {
 			fqdn := FormatHostnameForPrefixSearch(dHost, n.DestinationRule.GetObjectMeta().Namespace, n.DestinationRule.GetObjectMeta().ClusterName)
-			if versions, found := n.Services[fqdn.Service]; found {
-				if hasSubsets(n.DestinationRule) {
-					indexes := kubernetes.ValidateDestinationRulesSubsets([]kubernetes.IstioObject{n.DestinationRule}, fqdn.Service, versions)
```

Comment:
If we don't use `ValidateDestinationRulesSubsets`, do you think we can remove it? I think this is the only one usage of it.

Reply:
I'm not sure what you refer to "Why do you say that is missing one label?". The idea is to find a matching workload and a difference of even one label is enough to reject that offering. To my knowledge we don't see other than app == service == host configuration, at least all our examples and tests assume such. As such, it's a quick first check to make sure that it makes sense to check rest of the labels - improving filtering performance.

I'll change it to make use of the AppLabel and also remove unused code from the kubernetes package.